core/reactor_common.c

```diff
@@ -1845,7 +1845,12 @@ int process_args(int argc, char* argv[]) {
             }
             char* time_spec = argv[i++];
             char* units = argv[i++];
+
+            #ifdef BIT_32
+            duration = atol(time_spec);
```

Comment:
There is an issue here that a 32-bit representation of time in microseconds cannot exceed about 2,147 seconds, which is about 35 minutes. If a timeout is given, it cannot exceed 35 minutes.  Also, what will happen to application with no timeout when it runs for more than 35 minutes?

Reply:
From Arduino's perspective, the internal timer will overflow back to 0 after around ~70 minutes since its internal clock is tracked as an unsigned 32 bit number. I can do some mathematics for setting an overflow timer to resolve this, but I would need to work out whether we can convert int_32t values to uint_32t values without any issues.

Reply:
The same issue has been flagged here: https://github.com/icyphy/lf-buckler/issues/3

Should we use an alternative representation that consists of a tuple `(s, m)` where `s` represents the number of seconds and `m` represents the number of microseconds?

Reply:
Discussed this with @lhstrh at our meeting yesterday. Will probably have a separate pull request for handling overflow since this is an issue for all 32 bit boards. We will most likely need to spread the timer across two variables to remedy the problem at hand. Will work on this in a future PR.